mitodl/mitxonline
# Adds username normalization

#### Pre-Flight checklist


#### What are the relevant tickets?

#259 

#### What's this PR do?

Fixes #259.

This adds a `normalized_username` field to the users model and an API function to perform the normalization. It also updates the openedx `create_edx_user` call to pass the `normalized_username` field instead of the regular `username` field. 

#### How should this be manually tested?

The migrations to add the `normalized_username` field will automatically normalize the usernames. Because of that, it's pretty important that your user database does not contain accounts that will conflict. (Per #259, the `normalized_username` field is uniqued.) There is a management command to check for and fix usernames that would conflict that's in PR https://github.com/mitodl/mitxonline/pull/629.

After performing the migrations, create a user with a username using accented characters (áüô, etc.). The username should be created on the Open edX side with those characters converted to non-accented characters (auo, etc.). 

Some characters will _not_ be normalized and will instead be dropped. This includes ø so testing with that character will result in a normalized username without that character (for example, Søren will normalize to Sren).
